lrlex/src/lib/mod.rs

```diff
@@ -29,12 +29,15 @@ pub use crate::{
     lexer::{LRNonStreamingLexer, LRNonStreamingLexerDef, LexerDef, Rule},
 };
 
+use cfgrammar::Span;
+
 pub type LexBuildResult<T> = Result<T, LexBuildError>;
 
 /// Any error from the Lex parser returns an instance of this struct.
 #[derive(Debug)]
 pub struct LexBuildError {
     pub kind: LexErrorKind,
+    pub span: Span,
```

Comment:
This is interesting because we now have two ways of locating the error: `span` and `line`/`col`. The challenge that brings is that in order to make sense of `span` you need to have the input available, which perhaps suggests that `LexBuildError` should have a reference to the source input. Ironically that's quite easy because at the moment you need to move ownership of the source to the parser:

```rust
impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
    pub(super) fn new(src: String) -> LexBuildResult<LexParser<StorageT>> {
        ...
    }
}
```

In retrospect that should probably have been `src: &str`, but that's not directly important here.

So one possibility is that `LexBuildError` should be:

```rust
pub struct LexBuildError {
    pub kind: ...,
    span: Span,
    src: String
}
```

since it can then rederive `line`/`col` automatically. *But* that might not make sense in the context of LPS sorts of things... Any thoughts?

Reply:
Clearly not only did you have thoughts, but your later commits address them -- mea culpa! Still, it might be worth at least briefly thinking about whether we should carry around `src` or not. [I'm inclined to think "don't carry it around" but I haven't thought things through fully.]

Reply:
Yeah, I hadn't occurred to me to try and make it owned, so I haven't considered it to this point enough to have really formed an opinion.  :thinking:

Reply:
So thought about this, and my general thinking is that as you said it really doesn't make much sense in LSP sorts of contexts. There you'd have to keep a clone around for your own use, and then rely on the fact that spans between the two clones still correlate because they don't mutate the owned value.

One thing i'm wondering about `&str` wise is if it would be good to use `AsRef<str>`, (not sure it belongs in this, perhaps another time/patch?)

Reply:
I agree: let's keep `kind` and `span` and not add `src`.